### PR TITLE
[nrf fromlist] cmake: Zephyr module processing set modules path

### PR DESCRIPTION
**Pull request by tejlmand**
This PR is inspired by: https://github.com/NordicPlayground/fw-nrfconnect-nrf/pull/1576.
to minimize the need for knowing the path of other modules, as seen here:
https://github.com/hakonfam/fw-nrfconnect-nrf-1/blob/41d353f5e67c3dbc0501807b4ccc3cb28a4c1be2/cmake/mcuboot.cmake#L2

----

This commit introduces the variable:
ZEPHYR_${MODULE}_MODULE_DIR that can be used for modules to obtain
locations of other modules.

As example: a module which requires the knowledge of mcuboot con now
fetch this information using ZEPHYR_MCUBOOT_MODULE_DIR.

Signed-off-by: Torsten Rasmussen <Torsten.Rasmussen@nordicsemi.no>

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -467,20 +467,30 @@ add_subdirectory(drivers)
...
   foreach(module ${ZEPHYR_MODULES_TXT})
     # Match "<name>":"<path>" for each line of file, each corresponding to
     # one module. The use of quotes is required due to CMake not supporting
     # lazy regexes (it supports greedy only).
     string(REGEX REPLACE "\"(.*)\":\".*\"" "\\1" module_name ${module})
     string(REGEX REPLACE "\".*\":\"(.*)\"" "\\1" module_path ${module})
+
+    list(APPEND module_names ${module_name})
+
+    string(TOUPPER ${module_name} MODULEUP)
```
Comment (SebastianBoe) [inline CMakeLists.txt]: I find the \<var\>UP syntax hard to read, I'd prefer \<var\>_uppercase.

Comment (tejlmand) [reply CMakeLists.txt]: Fixed

**Comment by tejlmand** [comment]
> This is great, let's send it upstream.

> Can you please send this directly upstream?
> We are going to upmerge in the next few days

Done:
https://github.com/zephyrproject-rtos/zephyr/pull/21226

**Comment by mbolivar** [comment]
History in this repository has been rewritten. Please rebase this PR.

**Comment by tejlmand** [comment]
> Edit: actually, can't this simply be closed? We should get this in the next mergeup via [zephyrproject-rtos/zephyr#21226](https://github.com/zephyrproject-rtos/zephyr/pull/21226).

If next rebase is happening soon (which I assume it will), then agreed, we can simply close it.

**Comment by mbolivar** [comment]
> If next rebase is happening soon (which I assume it will), then agreed, we can simply close it.

I believe @anangl is beginning the next rebase soon.

**Comment by mbolivar** [comment]
> > If next rebase is happening soon (which I assume it will), then agreed, we can simply close it.
> 
> I believe @anangl is beginning the next rebase soon.

I'm sorry, I meant to say "mergeup", not "rebase".

**Comment by anangl** [comment]
> > > If next rebase is happening soon (which I assume it will), then agreed, we can simply close it.
> > 
> > 
> > I believe @anangl is beginning the next rebase soon.
> 
> I'm sorry, I meant to say "mergeup", not "rebase".

Yes, I started the upmerge procedure. A corresponding PR should appear soon.

**Comment by tejlmand** [comment]
> 'm sorry, I meant to say "mergeup", not "rebase".

Of course and so did I.
Stupid blind copying of words from message being replied 😄 

Closed. Should be part of next upmerge.

**Pull request closed without merging**